packages/botonic-core/src/hubtype-service.test.js

```diff
@@ -0,0 +1,23 @@
+import { HubtypeService } from './hubtype-service'
+
+describe('HubtypeService', () => {
+  test.each([
+    [{ user: { id: 'userId' } }],
+    [{ appId: 'appId' }],
+    [{ appId: '', user: { id: 'userId' } }],
+    [{ appId: 'appId', user: { id: undefined } }],
+  ])('init fails if user or appid are empty: %s', async args => {
+    const sut = new HubtypeService(args)
+
+    await expect(sut.init()).rejects.toEqual(
```

Comment:
Can you call init with no parameters? This is not TS but ES6, right? Getting confused here

Reply:
yes, it's ES6. We're migrating anyway to TS in #1419 (actually it will be moved to another PR)